### PR TITLE
[MINT-3020] Collaboration area page

### PR DESCRIPTION
**Pull request by patrickseguraoddball**
# MINT-3020

## Description

- Created page and route `/collaboration-area`
- Created/updated `Breadcrumbs` component to have default/preset breadcrumbs
  - Implemented this updated component across all of MINT
  - Includes all breadcrumb updates for Collaboration Area
- Added better support for underlining the current tab based on the parent route
  - `getActiveTab` util to identify tab that should be underlined, addresses custom/edge cases
- Added `ShareExportButton` component to render a select dropdown to share, export or remove model plan
  - Share or export will preselect the chosen tab when opened
- Left some commented out code in preparation for future tickets - ex: document, crtdl, teams etc.  This was ported over from the task list.  
- Updated emails to correctly include collaboration base routes
- Updated unit tests
- Updated cypress tests

## How to test this change

1.  Log in as both/either admin on base user
2. Click on any model plan from the home page tables (All models, My models, etc)
3. Verify that it navigates to the collaboration area component (if have edit access)
4. Verify share export model works as expected
5. Verify all breadcrumbs match figma

## PR Author Checklist

- [ ] I have provided a detailed description of the changes in this PR.
- [ ] I have provided clear instructions on how to test the changes in this PR.
- [ ] I have updated tests or written new tests as appropriate in this PR.
- [ ] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.

### CODE REVIEW

File: src/views/ModelPlan/CollaborationArea/index.tsx [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+              </Grid>
+
+              <Grid desktop={{ col: 3 }} className="margin-top-4">
+                <div className="display-flex flex-justify-end">
+                  <FavoriteIcon
+                    isFavorite={isFavorite}
+                    modelPlanID={modelID}
+                    updateFavorite={handleUpdateFavorite}
+                    isCollaborationArea
+                  />
```
Comment (garyjzhao) [inline src/views/ModelPlan/CollaborationArea/index.tsx:213]: FavoriteIcon component is missing the hover states that Natasha specified here: https://www.figma.com/design/smHRxJ9HIdywFV052vAc3Q/MINT-System-Library?node-id=524-205&t=Pk7FuMomhL7XqLfa-0

**Review by garyjzhao — CHANGES_REQUESTED** [review]
FE work looks great! Just one thing I noticed

File: src/views/ModelPlan/CollaborationArea/index.tsx (hunk 1 above)
Comment (patrickseguraoddball) [reply src/views/ModelPlan/CollaborationArea/index.tsx:213]: Nice catch thanks!

**Review by ClayBenson94 — CHANGES_REQUESTED** [review]
Just requesting changes to mark that we need to update the model plan status that just made it to main here!

https://github.com/CMSgov/mint-app/blob/6f6f7329f1a1220907a1587962709ea6351438a3/pkg/email/templates/model_plan_suggested_phase_body.html#L28-L34

**Review by garyjzhao — APPROVED** [review]
LGTM

**Pull request merged**